cartridge/stateboard-client.lua

```diff
@@ -0,0 +1,158 @@
+local checks = require('checks')
+local errors = require('errors')
+local netbox = require('net.box')
+
+local function acquire_lock(session, lock_args)
+    checks('stateboard_session', 'table')
+    assert(session.connection ~= nil)
+
+    local lock_acquired, err = errors.netbox_call(session.connection,
+        'acquire_lock', lock_args,
+        {timeout = session.call_timeout}
+    )
+    if lock_acquired == nil then
+        return nil, err
+    end
+
+    session.lock_acquired = lock_acquired
+    return lock_acquired
+end
+
+local function get_lock_delay(session)
+    checks('stateboard_session')
+    assert(session.connection ~= nil)
+
+    if session.lock_delay ~= nil then
+        return session.lock_delay
+    end
+
+    local lock_delay, err = errors.netbox_call(session.connection,
+        'get_lock_delay', nil,
+        {timeout = session.call_timeout}
+    )
+    if lock_delay == nil then
+        return nil, err
+    end
+
+    session.lock_delay = lock_delay
+    return lock_delay
+end
+
+local function set_leaders(session, updates)
+    checks('stateboard_session', 'table')
+    assert(session.connection ~= nil)
+
+    return errors.netbox_call(session.connection,
+        'set_leaders', {updates},
+        {timeout = session.call_timeout}
+    )
+end
+
+local function get_leaders(session)
+    checks('stateboard_session')
+    assert(session.connection ~= nil)
+
+    return errors.netbox_call(session.connection,
+        'get_leaders', nil,
+        {timeout = session.call_timeout}
+    )
+end
+
+local function is_locked(session)
+    checks('stateboard_session')
+    assert(session.connection ~= nil)
+
+    return session.connection:is_connected()
+        and session.lock_acquired
+end
+
+local function is_alive(session)
+    checks('stateboard_session')
+    assert(session.connection ~= nil)
+
+    return session.connection.state == 'initial'
```

Comment:
![image](https://user-images.githubusercontent.com/8830475/78269917-ddd7f400-7512-11ea-887d-2038cdce98ef.png)

Let's consider "alive" only as "connection:is_connected"

Reply:
Nope. We create session with `wait_connected = false` flag. "Initial" state is appropriate for the session to be counted as alive

Reply:
But there are also "connecting", "auth" states. And there is no difference between "initial" and these states.